core/src/databases/remote_databases/get_remote_database.rs

```diff
@@ -0,0 +1,21 @@
+use anyhow::Result;
+
+use crate::{
+    databases::remote_databases::remote_database::RemoteDatabase,
+    oauth::{client::OauthClient, credential::CredentialProvider},
+};
+
+use super::snowflake::SnowflakeRemoteDatabase;
+
+pub async fn get_remote_database(
```

Comment:
weird to create files for a static method. can we move that in remote_database.rs?

Reply:
No because we'll then have a circular dependency :/ 

`get_remote_database` needs snowflake, and `snowflake.rs` needs `remote_database.rs`

Reply:
No because we'll then have a circular dependency :/ 

`get_remote_database` needs snowflake, and `snowflake.rs` needs `remote_database.rs`

Reply:
I don't like it either tbh but not sure where to put this. Could be in `database.rs` I guess but not the most satisfying. 
Otherwise, `remote_databases::helpers` ?

Reply:
Hmmmm is that an issue? We have a bunch of similar pattern with model providers etc.. ?

Reply:
Let me see if it works 👍

Reply:
Ok my bad this works, I was being too javascript-brained